BLSQ/iaso
# POLIO-995: add budget step at start of workflow

we need to add a new budget step at the start of the workflow, to initiate it. It should be performed by ORPG

Related JIRA tickets : POLIO-995

## Self proofreading checklist


## Changes

- add new node and new transition in workflow json config: https://drive.google.com/drive/u/0/folders/1ua4yzyFOGayweQVNKP0hyBNDKhQrPlx1
- Update `Campaign`model and api with the corresponding field
- Add translations
- add new field to front-end
- Convert `BudgetForm` to TS

## How to test
- With Db configured for polio budgets and the workflow config linked above
- With your backend configured to use smtp bucket:

- Go to the budget tool: you should be able to add the new step, if your user is part of ORPG ops team
- ORPG team should receive the "submitter's email"
- UNICEF CO, WHO CO and GPEI Coordinator should receive the base email. Only the CO should have "quick buttons " in their email
- Go to the campaign page. Select the campaign you just updated, go to the budget tab: you should see the cahnges you just made
- Now go to another campaign (without data from budget tool): you should be able to fill in the "Risk Assessment completed" field

## Print screen / video


https://user-images.githubusercontent.com/38907762/234602975-40e53b87-516a-443f-91c1-9bffb2512092.mov

![Screenshot 2023-04-26 at 16 11 40](https://user-images.githubusercontent.com/38907762/234603029-dac3f62b-d8f2-42b1-ad76-c0b19ce683ff.png)



## Notes

The workflow configs have already been updated on staging and campaigns-staging. Only prod remains.
